core/store/src/trie/iterator.rs

```diff
@@ -206,7 +206,10 @@ impl<'a> TrieIterator<'a> {
     fn descend_into_node(&mut self, hash: &CryptoHash) -> Result<(), StorageError> {
         let (bytes, node) = self.trie.retrieve_node(hash)?;
         if let Some(ref mut visited) = self.visited_nodes {
-            visited.push(bytes.ok_or(StorageError::MissingTrieValue)?);
+            visited.push(bytes.ok_or(StorageError::MissingTrieValue(
```

Comment:
Just fyi this syntax is eager so the StorageError will alway be constructed, even if the `bytes` variable is an Ok. In this case it doesn't matter to much because it should be really cheap. I believe the lazy alternative is ok_else or something similar.